packages/app/src/cli/services/dev/urls.ts

```diff
@@ -89,14 +89,22 @@ export async function generateURL(config: Config, frontendPort: number): Promise
   return result.url
 }
 
-export function generatePartnersURLs(baseURL: string): PartnersURLs {
-  return {
-    applicationUrl: baseURL,
-    redirectUrlWhitelist: [
+export function generatePartnersURLs(baseURL: string, authCallbackPath?: string): PartnersURLs {
+  let redirectUrlWhitelist: string[]
+  if (authCallbackPath && authCallbackPath.length > 0) {
+    const cleanCallbackPath = authCallbackPath.startsWith('/') ? authCallbackPath : `/${authCallbackPath}`
```

Comment:
What about moving this logic to the schema definition. Zod, the package that we use under the hood, [has APIs](https://github.com/colinhacks/zod#preprocess) to process users' input before validating.